lib/core/src/Cardano/Wallet/Primitive/Types.hs

```diff
@@ -1173,6 +1177,37 @@ instance NFData StartTime
 newtype ProtocolMagic = ProtocolMagic Int32
     deriving (Generic, Show)
 
+{-------------------------------------------------------------------------------
+                                   Network
+-------------------------------------------------------------------------------}
+
+-- | Network-Time Protocol status.
+--
+-- TODO: Use a NTP-client to query some known NTP server and figure out by how
+-- much is machine's clock drifting.
+-- An existing NTP client implementation exists on `cardano-sl` and is fairly
+-- decoupled from the rest of the code:
+--
+-- https://github.com/input-output-hk/cardano-sl/blob/1a792d7cd0f0c93a0f0c28f66372bce3c3808dbd/networking/src/Ntp/Client.hs
+--
+-- The idea would be to extract that piece of code into a separate package that
+-- we could re-use to implement this functionality.
+data NtpStatus
+    = NtpUnavailable
+    | NtpPending
+    | NtpAvailable (Quantity "microsecond" Word32)
+    deriving (Generic, Eq, Show)
+
+-- | Status about protocol updates we've seen on the network.
```

Comment:
Is it possible to expand a little bit about what this is for? Who is supplying the protocol updates? When do they arrive? And so on... :)

Reply:
I wish I could :upside_down_face: ... 
Protocol updates are a network construct, they're part of the blockchain itself and Ouroboros. Although, we only see the proposals but not the approvals. So far, there's no way to know about approvals.